examples/guestbook/README.md

```diff
@@ -103,20 +128,96 @@ spec:
+
+```console
+$ kubectl create -f examples/guestbook/redis-master-service.yaml
+service "redis-master" created
+```
+
+Then check the list of services, which should include the redis-master:
+
+```console
+$ kubectl get services
+NAME              CLUSTER_IP       EXTERNAL_IP       PORT(S)       SELECTOR                            AGE
+redis-master      10.0.136.3       <none>            6379/TCP      app=redis,role=master,tier=backend  1h
+```
+
+This will cause all pods to see the redis master apparently running on <ip>:6379.  A service can map an incoming port to any `targetPort` in the backend pod.  Once created, the service proxy on each node is configured to set up a proxy on the specified port (in this case port 6379).
+
+`targetPort` will default to `port` if it is omitted in the configuration. For simplicity's sake, we omit it in the following configurations.
+
+The traffic flow from slaves to masters can be described in two steps, like so:
+
+  - A *redis slave* will connect to "port" on the *redis master service*
+  - Traffic will be forwarded from the service "port" (on the service node) to the  *targetPort* on the pod that the service listens to.
+
+For more details, please see [Connecting applications](../../docs/user-guide/connecting-applications.md).
+
+#### Finding a service
+
+Kubernetes supports two primary modes of finding a service— environment variables and DNS.
+
+The services in a Kubernetes cluster are discoverable inside other containers [via environment variables](../../docs/user-guide/services.md#environment-variables).
+
+An alternative is to use the [cluster's DNS service](../../docs/user-guide/services.md#dns), if it has been enabled for the cluster.  This lets all pods do name resolution of services automatically, based on the service name.
+
+This example has been configured to use the DNS service by default.
+
+If your cluster does not have the DNS service enabled, then you can use environment variables by setting the
+`GET_HOSTS_FROM` env value in both
+`examples/guestbook/redis-slave-controller.yaml` and `examples/guestbook/frontend-controller.yaml`
+from `dns` to `env` before you start up the app.
+(However, this is unlikely to be necessary. You can check for the DNS service in the list of the clusters' services by
+running `kubectl --namespace=kube-system get rc`, and looking for a controller prefixed `kube-dns`.)
+Note that switching to env causes creation-order dependencies, since services need to be created before their clients that require env vars.
+
+Second create the redis master pod in your Kubernetes cluster by running:
 
 ```console
 $ kubectl create -f examples/guestbook/redis-master-controller.yaml
-replicationcontrollers/redis-master
+replicationcontroller "redis-master" created
 ```
 
-The `replicationcontrollers/redis-master` line is the expected response to this operation.
+The `replicationcontroller "redis-master" created` line is the expected response to this operation.
```

Comment:
Remove "The `replicationcontroller "redis-master" created` line is the expected response to this operation." The output is much more clear than `replicationcontrollers/redis-master` thus we don't need to explain it.